cmd/kwild/root/root.go

```diff
@@ -112,6 +121,57 @@ func RootCmd() *cobra.Command {
 	return cmd
 }
 
+// parseExtensionFlags parses the extension flags from the command line and
+// returns a map of extension names to their configured values
+func parseExtensionFlags(args []string) (map[string]map[string]string, error) {
+	exts := make(map[string]map[string]string)
+	for i := 0; i < len(args); i++ {
+		if !strings.HasPrefix(args[i], "--") {
+			return nil, fmt.Errorf("expected extension flag, got %q", args[i])
+		}
+		// split the flag into the extension name and the flag name
+		// we intentionally do not use SplitN because we want to verify
+		// there are exactly 3 parts.
+		parts := strings.Split(args[i], ".")
+		if len(parts) != 3 {
+			return nil, fmt.Errorf("invalid extension flag %q", args[i])
+		}
+
+		extName := strings.TrimPrefix(parts[1], "--extension.")
```

Comment:
`parts[1]` doesn't need formatting

Reply:
btw, I think `parseExtensionFlags` should check `parts[0]` equals `--extension`

Reply:
ok

Reply:
I now check that the arg starts with `--extension.`